### PR TITLE
Adds support for "Still watching" prompt to prevent users from idling when watching stream

### LINKED ISSUE #8170: Add a Check to Ensure Users Are Actively Watching Camera Feeds
**Is your feature request related to a problem? Please describe.**

When a patient camera feed page is turned on and left stale, it continues to play until the user closes the page, which can lead to unnecessary resource usage.

**Describe the solution you'd like**

Implement a feature that triggers a pop-up after the camera feed has been running for 15 minutes, asking the user "Are you still watching

- If the user does not respond within one minute, pause the feed.
- If the user responds "No", pause the feed.
- If the user responds "Yes", restart the 15-minute timer and show the pop-up again after the next 15 minutes.

Note : 

- During the one-minute response period, do not freeze the feed.
- On the live feed page that displays multiple devices, perform the validation check centrally for the entire page rather than individually for each camera feed.

CC : @gigincg 

_Issue generated based on engineering discussion 15/07/2024_

### PR DESCRIPTION
**Pull request by rithviknishad**
## Proposed Changes

- Fixes #8170
- Shows a prompt to confirm whether the user is still watching the stream with a linear backoff of 3 minutes base saturated (at most 12 minutes). Once timed-out, unmounts the child component.

(Set a faster timeout in screen recording for demo purpose)

https://github.com/user-attachments/assets/cc98d995-e1e3-4d40-864e-84b2dd162f8d

@ohcnetwork/code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA

### CODE REVIEW

**Comment by rithviknishad** [comment]
Adding i18n...

**Review by bodhish — CHANGES_REQUESTED** [review]
Let's make the time configurable via env;

**Comment by rithviknishad** [comment]
> Let's make the time configurable via env;

How about through [config.json](https://care.ohc.network/config.json) instead? (All whitelablling envs are present there)

**Comment by rithviknishad** [comment]
Also @bodhish the configuration should be for the base delay for the linear backoff right?

**Comment by bodhish** [comment]
I'm not sure where we have kept the end, but let's keep it at a unified place.

Also set a sensible default, if the env is not present, the default should be taken

**Review by sainak — APPROVED** [review]

File: src/CAREUI/misc/RemainingTime.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      setRemaining(time - new Date().getTime());
+    }, 1000);
+
+    return () => {
+      clearInterval(interval);
+    };
+  }, [time]);
+
+  return `${Math.max(remaining / 1e3, 0).toFixed(0)}s.`;
+}
```
Comment (shivankacker) [inline src/CAREUI/misc/RemainingTime.tsx]: Can we use the `seconds` returned in `useTimer` hook that was recently added. Something like`${time_to_wait-seconds}s left`

**Review by shivankacker — CHANGES_REQUESTED** [review]

File: src/CAREUI/misc/RemainingTime.tsx (hunk 1 above)
Comment (rithviknishad) [reply src/CAREUI/misc/RemainingTime.tsx]: Done

**Comment by nihal467** [comment]
@rithviknishad the camera authentication frontend PR is buggy so keep this PR on hold since the camera feed is not working in develop

**Comment by rithviknishad** [comment]
@nihal467 This can be tested without the feed being live. The still watching component wrapper is agnostic of where it's used. It just wraps over any component and observes for touches within that wrapped component.

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Review by shivankacker — APPROVED** [review]

**Review by bodhish — APPROVED** [review]
LGTM

**Pull request merged**